src/main/java/com/mraof/minestuck/blockentity/ComputerBlockEntity.java

```diff
@@ -198,55 +208,45 @@ public ComputerReference createReference()
 	}
 	
 	@Override
-	public boolean getClientBoolean(String name)
-	{
-		return getData(0).getBoolean(name);
-	}
-	
-	@Override
-	public boolean getServerBoolean(String name)
+	public boolean getProgramBoolean(int id, String name)
 	{
-		return getData(1).getBoolean(name);
+		return getData(id).getBoolean(name);
 	}
 	
 	@Override
-	public void putClientBoolean(String name, boolean value)
+	public void putProgramBoolean(int id, String name, boolean value)
 	{
-		getData(0).putBoolean(name, value);
+		getData(id).putBoolean(name, value);
 		setChanged();
 		markBlockForUpdate();
 	}
 	
 	@Override
-	public void putServerBoolean(String name, boolean value)
+	public void clearConnectedClient()
 	{
-		getData(1).putBoolean(name, value);
+		getData(1).putString("connectedClient", "");
 		setChanged();
 		markBlockForUpdate();
 	}
 	
 	@Override
-	public void clearConnectedClient()
+	public void putProgramMessage(int id, String message)
 	{
-		getData(1).putString("connectedClient", "");
+		latestmessage.put(id, message);
 		setChanged();
 		markBlockForUpdate();
 	}
 	
 	@Override
-	public void putClientMessage(String message)
+	public Theme getTheme()
 	{
-		latestmessage.put(0, message);
-		setChanged();
-		markBlockForUpdate();
+		return theme;
 	}
 	
 	@Override
-	public void putServerMessage(String message)
+	public void setTheme(Theme theme)
 	{
-		latestmessage.put(1, message);
-		setChanged();
-		markBlockForUpdate();
+		this.theme = theme;
```

Comment:
Since this changes data that should be serialized, you should make sure to call `setChanged()`. I guess you may also need to call `markBlockForUpdate()` to make sure it gets synced to client-side?